shared/devices/index.js

```diff
@@ -1,182 +1,151 @@
-        </Text>
-      </Box>
-    )}
-  </Box>
-)
+class Devices extends React.PureComponent<Props & FloatingMenuParentProps, State> {
+  static defaultProps = {_stateOverride: null}
+  state = {revokedExpanded: this.props._stateOverride ? this.props._stateOverride.revokedExpanded : false}
+
+  componentDidMount() {
+    this.props.loadDevices()
+  }
 
-const DeviceRow = RowConnector(({isCurrentDevice, name, isRevoked, icon, showExistingDevicePage}) => (
-  <ClickableBox onClick={showExistingDevicePage} style={{...stylesCommonRow, alignItems: 'center'}}>
-    <Box key={name} style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
-      <Icon type={icon} style={isRevoked ? {marginRight: 16, opacity: 0.2} : {marginRight: 16}} />
-      <Box style={{...globalStyles.flexBoxColumn, flex: 1, justifyContent: 'flex-start'}}>
-        <Text style={textStyle(isRevoked)} type="BodySemiboldItalic">
-          {name}
-        </Text>
-        {isCurrentDevice && <Text type="BodySmall">Current device</Text>}
-      </Box>
-    </Box>
-  </ClickableBox>
-))
+  _toggleExpanded = () => this.setState(p => ({revokedExpanded: !p.revokedExpanded}))
 
-class _Devices extends PureComponent<Props & FloatingMenuParentProps> {
   _renderRow = (index, item) =>
     item.type === 'revokedHeader' ? (
       <RevokedHeader
         key="revokedHeader"
-        expanded={this.props.showingRevoked}
-        onToggleExpanded={this.props.onToggleShowRevoked}
+        expanded={this.state.revokedExpanded}
+        onToggleExpanded={this._toggleExpanded}
       />
     ) : (
-      <DeviceRow key={item.id} deviceID={item.id} />
+      <DeviceRow key={item.id} deviceID={item.id} firstItem={index === 0} />
     )
 
   render() {
     const items = [
-      ...this.props.deviceIDs.map(id => ({id, key: id, type: 'device'})),
-      {key: 'revokedHeader', type: 'revokedHeader'},
-      ...(this.props.showingRevoked
-        ? this.props.revokedDeviceIDs.map(id => ({id, key: id, type: 'device'}))
-        : []),
+      ...this.props.items,
+      ...(this.props.items.length ? [{key: 'revokedHeader', type: 'revokedHeader'}] : []),
+      ...(this.state.revokedExpanded ? this.props.revokedItems : []),
+    ]
+
+    const menuItems = [
+      {onClick: this.props.addNewPhone, title: 'New phone'},
+      {onClick: this.props.addNewComputer, title: 'New computer'},
+      {onClick: this.props.addNewPaperKey, title: 'New paper key'},
     ]
 
     return (
-      <Box2 direction="vertical" fullHeight={true} fullWidth={true}>
+      <Common.Box2 direction="vertical" fullHeight={true} fullWidth={true}>
         <DeviceHeader
           setAttachmentRef={this.props.setAttachmentRef}
           onAddNew={this.props.toggleShowingMenu}
           waiting={this.props.waiting}
         />
-        <List items={items} renderItem={this._renderRow} />
+        {this.props.waiting && <Common.ProgressIndicator style={styles2.progress} />}
```

Comment:
This is rendering in the upper left corner of the window, on top of the tab bar
<img width="85" alt="image" src="https://user-images.githubusercontent.com/11968340/43465002-60b54e5e-94aa-11e8-9bba-fc8f5396867c.png">